validation/src/index.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+const path = require('path');
+const { validateAllSchemas } = require('./schemaValidation');
+const { validateURLs } = require('./validateURLs');
+const { validateAllWhitelistedFiles } = require('./validateWhitelistedFiles');
+const { validateAllConfigFiles } = require('./validateConfigFiles');
+const { validateConfigFilePaths } = require('./validateConfigFilePaths');
+const config = require('../config');
+
+const validate = async () => {
+	// Get all modified files
+	const allChangedFiles = process.argv.slice(2);
+
+	// Get all app dir added or modified inside network dirs
+	const changedAppDirs = new Set();
+	for (let i = 0; i < allChangedFiles.length; i++) {
+		const dir = allChangedFiles[i].split('/').slice(0, 2).join('/');
+		if (dir.trim() && config.knownNetworks.includes(dir.split('/')[0])) {
+			changedAppDirs.add(path.resolve(dir));
+		} else {
+			throw new Error('Added/Updated a file not belonging to network directory.');
```

Comment:
printing the filepath will help with debugging

```suggestion
			throw new Error(`Added/Updated a file not belonging to network directory. filepath:${allChangedFiles[i]}`);
```